array.c

```diff
@@ -2493,6 +2493,7 @@ ary_enum_length(VALUE ary, VALUE args, VALUE eobj)
  *  Allows the array to be modified during iteration:
  *    a = [:foo, 'bar', 2]
  *    a.each {|element| puts element; a.clear if element.to_s.start_with?('b') }
```

Comment:
I would remove these 3 lines, AFAIK they were never explicitly supported (and same below).
The documentation should not show an example of something that should be avoided.

Reply:
BTW that example was added by in #3216
@BurdetteLamar FYI, mutating while iterating is explicitly discouraged by matz: http://blade.nagaokaut.ac.jp/cgi-bin/scat.rb/ruby/ruby-core/23633

Reply:
> I would remove these 3 lines, AFAIK they were never explicitly supported (and same below). The documentation should not show an example of something that should be avoided.

I'm fine with removing all these.